pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -592,6 +592,22 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext, dest destinationBitset) {
 	ReportDataSecAtRestInfo.Status, ReportDataSecAtRestInfo.Info =
 		vault.GetOperationalInfo(log)
 	ReportDeviceInfo.DataSecAtRestInfo = ReportDataSecAtRestInfo
+	if len(ReportDataSecAtRestInfo.VaultList) > 0 {
+		// We assume we care about the first one since normally only one
+		first := ReportDataSecAtRestInfo.VaultList[0]
```

Comment:
But we still assume it can be a list when getting info here:
https://github.com/lf-edge/eve/blob/e6c396103577ef486001ec3d99072c2b873ea37d/pkg/pillar/cmd/zedagent/reportinfo.go#L1173
Should we add an assertion or comment on why we assume it is a list of one element?

Reply:
Done.